kv/txn_coord_sender.go

```diff
@@ -587,22 +545,31 @@ func (tc *TxnCoordSender) sendOne(ctx context.Context, call proto.Call) {
 // inconsistencies are found.
 // It is checked that the individual call does not have a User, UserPriority
 // or Txn set that differs from the batch's.
-func updateForBatch(aHeader *proto.RequestHeader, bHeader proto.RequestHeader) error {
+func updateForBatch(args proto.Request, bHeader proto.RequestHeader) error {
 	// Disallow transaction, user and priority on individual calls, unless
 	// equal.
+	aHeader := args.Header()
 	if aHeader.User != "" && aHeader.User != bHeader.User {
 		return util.Error("conflicting user on call in batch")
 	}
 	if aPrio := aHeader.GetUserPriority(); aPrio != proto.Default_RequestHeader_UserPriority && aPrio != bHeader.GetUserPriority() {
 		return util.Error("conflicting user priority on call in batch")
 	}
-	if aHeader.Txn != nil && !aHeader.Txn.Equal(bHeader.Txn) {
-		return util.Error("conflicting transaction in transactional batch")
-	}
-
 	aHeader.User = bHeader.User
 	aHeader.UserPriority = bHeader.UserPriority
-	aHeader.Txn = bHeader.Txn
+	// Only allow individual transactions in a batch if
```

Comment:
From discussion with @spencerkimball: It's weird that `ResolveIntent` even uses the header transaction, since it is not really a part of that transaction. Instead of allowing requests from multiple transactions to appear in the same batch, it's probably better to just add a Transaction field to ResolveIntentRequest so it doesn't use the header transaction at all (and then we can preserve the invariant that all requests in a batch have the same transaction)